pkg/api/graph/graph.go

```diff
@@ -404,6 +447,10 @@ func (g Graph) SubgraphWithNodes(nodes []graph.Node, fn EdgeFunc) Graph {
 	for _, node := range nodes {
 		out.internal.AddNode(node)
 	}
+	fmt.Println("Edges under review:")
+	for _, edge := range g.internal.Edges() {
```

Comment:
The problem seems to be when adding reverse edges. `g.internal.Edges` works fine before, after it returns the fixed number of edges it was sending before, now though the returned set get messed up with reverse edges.